remote_control.py

```diff
@@ -110,3 +120,23 @@ def constrain(self, val, min_val, max_val):
     def stop(self):
         logging.info("RC challenge stopping")
         self.killed = True
+
+    def dither(self, counter, speed):
+        if self.deadband < abs(speed) < self.slow_speed:
+            speed = int(speed - math.copysign(1, speed))
+            counter += 1
+            if counter < self.boost_cycles:
+                speed = speed + int(math.copysign(self.slow_speed, speed))
+            if counter > (self.boost_cycles + self.boost_dwell):
+                counter = 0
+        return (counter, speed)
+
+    def exp(self, demand, exp):
+    #fucntion takes a demand speed from -1 to 1 and converts it to a response value
+    # with an exponential function. exponential is -inf to +inf, 0 is linear
+        if exp < 0:

```

Comment:
Python has a nice one liner syntax for this in the form `a if condition else b`, so could be written more succinctly as:
`exp = 1/(1 + abs(exp)) if exp < 0 else exp + 1`

Reply:
btw - not everyone digs that syntax, as (among other reasons) the order of the arguments differs from other languages. Those people are idiots. It makes sense because it's almost exactly how you would describe the operation in natural language:
"make the thing equal a if condition is true, otherwise, make the thing equal b"